src/net.rs

```diff
@@ -42,7 +42,12 @@ impl Reactor {
     pub fn register(&self, sys: sys::TcpStream) -> io::Result<TcpStream> {
         sys.set_nonblocking(true)?;
         let mut sys = mio::net::TcpStream::from_std(sys);
-
+        let local_addr = sys.local_addr().unwrap();
+        let peer_addr = sys.peer_addr().unwrap();
+        let connection_info = ConnectionInfo {
```

Comment:
I don't think we need to store this on the `TcpStream`. We can just get it in the server loop.